### PR TITLE
#3936: Add a general comment field to person

### LINKED ISSUE #3936: Add a general comment field to person entity[1]
**Issue description is outdated, please ignore everything except the development specifications**

### Bug Description

If you converting a contact into a case the general comment is not transferred and must copied manually.

### Steps to Reproduce
1. write something into the general comment
grafik

2. transfer the contact into a case, the comment is empty
grafik

### Expected Behavior

The comment should be transferred into the case.

### System Details

* SORMAS version: 1.54.0-SNAPSHOT
* Server URL: https://test.sormas.netzlink.com
* User Role: Contact Supervisor

### Development Specifications
* Add a new `additionalDetails` field to persons (caption: "General comment"; `@SensitiveData` annotation)
* Add this field to the bottom of the person form and fragment (similar to how it's displayed for cases, i.e. don't show the caption of the field, but add a heading with the caption above it)
* Vaadin app: Create a `ResizableTextAreaWrapper` that allows users to extend the size of the new field; please have a look at `TextFieldWithMaxLengthWrapper` that does this (used for event description) - however, the new field should not have a max length

### PR DESCRIPTION
**Pull request by ChristopherRiedel**
closes #3936

### CODE REVIEW

**Review by alexcaruntu-vita — APPROVED** [review]

File: sormas-api/src/main/java/de/symeda/sormas/api/person/PersonDto.java [hunk 1]
```diff
@@ -883,6 +888,15 @@ public void setCitizenship(CountryReferenceDto citizenship) {
 		this.citizenship = citizenship;
 	}
 
+	@Column(length = COLUMN_LENGTH_BIG)
```
Comment (MateStrysewske) [inline sormas-api/src/main/java/de/symeda/sormas/api/person/PersonDto.java:891]: We don't need column definitions in DTOs, move this to `Person` in sormas-backend instead and use annotation `@Column(columnDefinition = "text")`

File: sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/DatabaseHelper.java [hunk 2]
```diff
@@ -2136,6 +2136,9 @@ public void onUpgrade(SQLiteDatabase db, ConnectionSource connectionSource, int
 				getDao(Location.class).executeRaw("ALTER TABLE location ADD COLUMN contactPersonPhone text;");
 				getDao(Location.class).executeRaw("ALTER TABLE location ADD COLUMN contactPersonEmail text;");
 
+				case 296:
+					currentVersion = 297;
```
Comment (MateStrysewske) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/DatabaseHelper.java:2140]: Wrong `currentVersion` assignment, needs to be identical to the case

File: sormas-app/app/src/main/java/de/symeda/sormas/app/backend/person/Person.java [hunk 3]
```diff
@@ -182,6 +183,8 @@
 	private Country birthCountry;
 	@DatabaseField(foreign = true, foreignAutoRefresh = true, maxForeignAutoRefreshLevel = 3)
 	private Country citizenship;
+	@Column(length = COLUMN_LENGTH_BIG)
```
Comment (MateStrysewske) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/backend/person/Person.java:186]: Use `@Column(columnDefinition = "text")` instead

File: sormas-backend/src/main/resources/sql/sormas_schema.sql [hunk 4]
```diff
@@ -7195,4 +7195,9 @@ ALTER TABLE labmessage ADD COLUMN pathogentest_id BIGINT;
 ALTER TABLE labmessage ADD CONSTRAINT fk_labmessage_pathogentest FOREIGN KEY(pathogentest_id) REFERENCES pathogentest(id) ON UPDATE NO ACTION ON DELETE NO ACTION;
 
 INSERT INTO schema_version (version_number, comment) VALUES (362, '[DEMIS Interface] visualize respective lab messages in sample and pathogen test sections #4853');
+
+-- 2021-04-12 Add additional details to person #3936
+ALTER TABLE person ADD COLUMN additionaldetails varchar(4096);
```
Comment (MateStrysewske) [inline sormas-backend/src/main/resources/sql/sormas_schema.sql:7200]: Use `text` instead of varchar (same in the history table)

**Review by MateStrysewske — CHANGES_REQUESTED** [review]

**Review by MateStrysewske — CHANGES_REQUESTED** [review]
* Add the @Column annotation to Person.java in sormas-backend
* Increase maxLines in the person edit fragment (mobile app) to 6
* Remove the maxLines parameter from the person read fragment (mobile app), otherwise it's impossible to display the whole text if it is too long

**Comment by syntakker** [comment]
Removing `maxLines` in `ControlTextReadField` defaults to `maxLines = 2`, so `maxLine` is set to 4000 in the read field instead.

**Review by MateStrysewske — APPROVED** [review]

**Pull request merged**